stage1/init/kvm.go

```diff
@@ -134,6 +130,14 @@ func doBindMount(source, destination string, readOnly bool, recursive *bool) err
 		return err
 	}
 
+	// Linux can't bind-mount with readonly in a single operation, so remount +ro
+	if readOnly {
+		if err := syscall.Mount("", destination, "bind", syscall.MS_REMOUNT|syscall.MS_RDONLY|syscall.MS_BIND, ""); err != nil {
```

Comment:
I think you should use `flags` here, otherwise you may be dropping the `MS_REC` flag.